src/renate/updaters/model_updater.py

```diff
@@ -187,10 +186,14 @@ def teardown(self, trainer: Trainer, pl_module: LightningModule, stage: str) ->
                 self.teardown(trainer, pl_module, stage)
             elif learner_state_path.exists() and learner_state_path.is_file():
                 # This a normal file. We strip the model of any wrappers and save that.
-                state_dict = torch.load(learner_state_path)["state_dict"]
-                out_sd = {k.replace("_model.", "", 1): v for k, v in state_dict.items()}
-                # Replace only 1 instance because we have to load it into RenateModule.
+                learner_state = torch.load(learner_state_path)
+                out_sd = {
+                    k.replace("_model.", "", 1): v for k, v in learner_state["state_dict"].items()
+                }  # Replace only 1 instance because we have to load it into RenateModule.
                 torch.save(out_sd, defaults.model_file(self.dirpath))
+                # Remove model from learner checkpoint
+                learner_state["state_dict"] = {}
```

Comment:
Why empty? `learner_state.pop("state_dict")` instead?

Reply:
If it is empty, loading with `strict=False` will make basically no changes. The key must exist since lightning explicitly accesses it.